auth/auth.go

```diff
@@ -116,23 +116,59 @@ func (c *Credentials) JSON() []byte {
 
 // ProjectID returns the associated project ID from the underlying file or
 // environment.
-func (c *Credentials) ProjectID() string {
-	return c.projectID
+func (c *Credentials) ProjectID(ctx context.Context) (string, error) {
```

Comment:
I'm wondering if `Credentials` could keep the original methods here as simple getter convenience wrappers for the new getters. These simple versions would pass in nil `ctx` (which panics if `ctx` is used) and they would also panic if the provider would return a non-nil `err`. In other words, these wrappers would panic if a networked impl was used, but would be safe (and convenient!) when `StaticCredentialsProperty` is used.

Reply:
But tbh, I haven't really looked at when these convenient "static" getters could actually be used...

Reply:
I think I would rather have one or the other and not both. And being this is more of a foundations library that other projects build on-top-of it feels more correct to have the errors. Anyone using the API can always throw the errors away if they truly don't want to check them, but I would rather leave that up to them. In reality, I don't think these methods get called too much, at least they have not been by our libraries historically. (This does change a bit with universe domain). We are for sure trading convenience here -- but I am personally okay with that over correctness for an authentication library.

Reply:
SGTM